activation/poet.go

```diff
@@ -152,6 +153,19 @@ func NewHTTPPoetClient(server types.PoetServer, cfg PoetConfig, opts ...PoetClie
 		opt(poetClient)
 	}
 
+	resp, err := poetClient.info(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.PhaseShift.AsDuration() != cfg.PhaseShift {
+		poetClient.logger.Warn("getting info about poet service configuration",
+			zap.Duration("server: phase shift", resp.PhaseShift.AsDuration()),
+			zap.Duration("config: cycle gap", cfg.PhaseShift),
+		)
+		return nil, ErrPhaseShiftMismatch
+	}
```

Comment:
I'm not sure if we want to query the poet service here. The poets have occasional downtimes. What if the poet service is unavailable at the moment the node starts? The node would shut down immediately.

Reply:
I think it is OK to query the PoET during start up if it is available to quickly detect a possible misconfiguration, but yeah if the PoET doesn't respond we should not exit immediately but instead maybe only log a warning that the PoET in question isn't online.

Are we checking before submitting that the PoET has the same phaseshift as the node?

Reply:
No, we just submitting challenge

Reply:
How about:
- query once during startup and
   - exit if phase isn't the same 
   - log a warning if poet doesn't respond, but otherwise continue normally
- query again (every poet) before submit and
   - log an error if phase is different

wdyt @ConvallariaMaj @poszu ?

Reply:
I thought about it too, but do not like it really much, because than logic is smeared between different layers and it is not obvious.

Reply:
Another way, avoiding additional querying /v1/info could be to return round information on /v1/submit, but IDK if this is a good idea :shrug:.

Anyway, a poet that is down (possibly temporarily) mustn't prevent a node from starting.

Reply:
> Anyway, a poet that is down (possibly temporarily) mustn't prevent a node from starting.

I agree, but should we log a warning to the user in your opinion if the PoET cannot be queried during startup or just ignore if it's not responding?

Reply:
Ok, let's move it there than